tests/telemetry/test_track.py

```diff
@@ -1,38 +1,45 @@
-import threading
+import json
+import time
+from dataclasses import asdict
+from typing import Any
 from unittest.mock import patch
 
 import pytest
 import sklearn
 
 import mlflow
 from mlflow.environment_variables import MLFLOW_DISABLE_TELEMETRY
-from mlflow.telemetry.client import TelemetryClient, get_telemetry_client, set_telemetry_client
-from mlflow.telemetry.schemas import APIStatus, AutologParams, Record
+from mlflow.telemetry.client import (
+    TelemetryClient,
+    get_telemetry_client,
+    set_telemetry_client,
+)
+from mlflow.telemetry.schemas import APIStatus, AutologParams
 from mlflow.telemetry.track import track_api_usage
 from mlflow.telemetry.utils import is_telemetry_disabled
 
+from tests.telemetry.helper_functions import wait_for_telemetry_threads
+
 
 def full_func_name(func):
     return f"{func.__module__}.{func.__qualname__}"
 
 
-def wait_for_telemetry_threads(timeout: float = 5.0):
-    """Wait for telemetry threads to finish to avoid race conditions in tests."""
-    for thread in threading.enumerate():
-        if thread != threading.main_thread() and thread.name == "add_telemetry_record":
-            thread.join(timeout=timeout)
+def extract_record(data: str) -> dict[str, Any]:
+    return json.loads(data["data"])
 
 
-def test_track_api_usage():
-    telemetry_client = get_telemetry_client()
-    assert telemetry_client.records == []
+def test_track_api_usage(mock_requests):
+    assert len(mock_requests) == 0
 
     @track_api_usage
     def succeed_func():
+        time.sleep(0.1)
```

Comment:
why do we need sleep?

Reply:
Oh sorry this was to make sure `duration_ms` field > 0 😅

Reply:
Do we need to sleep 100ms?